src/test/run-pass/impl-trait/nested.rs

```diff
@@ -0,0 +1,8 @@
+// run-pass
+#![allow(dead_code)]
+use std::borrow::Borrow;
+use core::fmt::Debug;
+
+fn test(_: impl Borrow<impl Debug>) {}
```

Comment:
This should definitely not be allowed as written here; There's even a comment saying so in https://github.com/rust-lang/rust/pull/57730/files#diff-1b4892578c33275e3b1ff1fe43b7f2f9.

Instead, this should be moved to [`test/ui/impl-trait/where.rs`](https://github.com/rust-lang/rust/blob/e10cecc25e987487ab8b06d34b1d84bda551a0b6/src/test/ui/impl-trait/where-allowed.rs) as an example of what isn't allowed.

Reply:
Well not exactly `impl A<impl B>`... it would be nice to have such a test.